### ISSUE TITLE
Precedence of the logical operations is not rendered correctly

### ISSUE THREAD

**Issue by mathematicswizard** [cortex-js/compute-engine#156]
## Description

The Compute Engine doesn't recognize the precedence of the logical operations.

### Steps to Reproduce

Write some propositions with logical connectives in a mathfield, such as:
3=4\vee 7=8
A\subseteq B\wedge\emptyset\subset B

### Actual Behavior

These propositions renders to the following MathJSON:
["Equal",3,["And",["Error",["ErrorCode","'incompatible-domain'","Booleans","PositiveIntegers"],4],["Error",["ErrorCode","'incompatible-domain'","Booleans","PositiveIntegers"],5]],7]
["SubsetEqual","A",["Subset",["And","B",["Error",["ErrorCode","'incompatible-domain'","Booleans","Sets"],"EmptySet"]],"B"]]

### Expected Behavior

["Or",["Equal",3,4],["Equal",7,8]]
["And",["SubsetEqual","A","B"],["Subset","EmptySet","B"]]

An error occurs because the Compute Engine reads all these propositions as **simple** propositions and not compound ones. So it find the first relational symbol and expects to find a noun to the left of it and a noun to the right it.

To note - adding parentheses "solves" the problem and the MathJSON is as above. But the paraentheses in these cases a superfluous - even without them these are still WFF's; these propositions cannot be interpreted in any other way - that is why the parentheses are not needed, and the Compute Engine should have read them accordingly.

### Environment

I have never checked if this worked in the past. Did this with the latest version of mathlive in the demo page and adding to the developer console this code:
formula._mathfield.getValue("math-json")

**Comment by mathematicswizard** [comment]
Thank you for your quick response with 2 logical connectives of implication and logical equivalence.

The term expression is ambiguous and in FOL and SOL which creates a universal language for every subject of mathematics we use the terms noun and proposition. You are right that \neg expect an atomic... but proposition and not an atomic noun. That's why you do not need parentheses with something like \neg1=1 because the most immediate atomic proposition is 1=1. 

In the same way:
$$ \neg\forall x.P\left(x\right) $$

Is rendered right now in computeengine as:
["Not", ["ForAll", "x", ["Predicate", "P", "x"]]]

In standard FOL we don't have equality sign (when we do we call it FOL+equality) and we could have had a binary Predicate we would interpret as equality (Let's call it P). and so we would have for example something like P(1,1). If we use \neg P(1,1) we wouldn't need parentheses.

Another example:
$$\neg 2\in\mathbb{R}$$

There is no other way to interpret this and so no parentheses needed. In fact we never need parentheses after negation, unless we want to negate a more complex proposition (for example \neg(p\vee q)).

Regarding the reference to programming languages - they have a very different syntax from mathematics. !1=1 has a meaning because 1 is another term for T. This doesn't happen in mathematics (usually). Also, in some programming languages when you try to use !x and x is in an int it will throw an error.

In conclusion - negation should also refer to the most immidiate **proposition** following it, not any noun.
